code/modules/antagonists/bloodsuckers/powers/targeted/hecata.dm

```diff
@@ -0,0 +1,203 @@
+//Hecata, the unified clan of death.
+//Focuses more on their vassals than other clans.
+//Has to rely more on vassals for ranks and blood, as they cannot use the blood altar for leveling up and cannot silent feed.
+//In exchange, they can raise the dead as temporary vassals to do their bidding, or permanently revive dead existing vassals.
+//In addition, they can summon Wraiths (shades) around them for both offense and defense
+//And can send messages to vassals anywhere globally via Dark Communion.
+//In addition, raising the dead with Necromancy turns them into Sanguine Zombies
+//They are pseudo zombies, which have high punch damage and special resistances, but aren't infectious nor can they use guns.
+/datum/action/cooldown/bloodsucker/targeted/hecata
+	purchase_flags = HECATA_CAN_BUY
+	background_icon = 'icons/mob/actions/actions_hecata_bloodsucker.dmi'
+	button_icon = 'icons/mob/actions/actions_hecata_bloodsucker.dmi'
+	active_background_icon_state = "hecata_power_on"
+	base_background_icon_state = "hecata_power_off"
+
+/datum/action/cooldown/bloodsucker/targeted/hecata/necromancy
+	name = "Necromancy"
+	button_icon_state = "power_necromancy"
+	desc = "Raise the dead as temporary vassals, or revive a dead vassal as a zombie permanently. Temporary vassals last longer as this ability ranks up."
+	power_explanation = "Necromancy:\n\
+		Click on a corpse in order to attempt to resurrect them.\n\
+		Non-vassals will become temporary zombies that will follow your orders. Dead vassals are also turned, but last permanently.\n\
+		Temporary vassals tend to not last long, their form quickly falling apart, make sure you set them out to do what you want as soon as possible.\n\
+		Vassaling people this way does not grant ranks. In addition, after their time is up they will die and no longer be your vassal."
+	check_flags = BP_CANT_USE_IN_TORPOR|BP_CANT_USE_IN_FRENZY|BP_CANT_USE_WHILE_UNCONSCIOUS
+	power_flags = BP_AM_STATIC_COOLDOWN
+	bloodcost = 35
+	cooldown_time = 45 SECONDS
+	target_range = 1
+	prefire_message = "Select a target."
+	
+/datum/action/cooldown/bloodsucker/targeted/hecata/necromancy/CheckValidTarget(atom/target_atom)
+	. = ..()
+	if(!.)
+		return FALSE
+	return isliving(target_atom)
+/datum/action/cooldown/bloodsucker/targeted/hecata/necromancy/CheckCanTarget(atom/target_atom)
```

Comment:
If you know this is Carbon (commented by in the proc), you can just path it ``mob/living/carbon/target_atom``.